### DISCUSSION TITLE
Green-Lagrange strain with Neo-Hookean model

### DISCUSSION THREAD

**Discussion by rh201** [idaholab/moose#23503] [Q&A General] [answered]
Hi MOOSEers,

I want to calculate Green-Lagrange strain with Neo-Hookean model, and I found this page [https://mooseframework.inl.gov/source/materials/ADComputeGreenLagrangeStrain.html](url), but it's used for St. Venant-Kirchhoff hyper elasticity model. Are there any modules/examples related to Neo-Hookean model? Thanks!

**Comment by hugary1995** [comment, marked as answer]
Yes, if you've searched within the repo or the website, this class and/or its documentation should pop up https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/materials/lagrangian/ComputeNeoHookeanStress.C

Note that Neo-Hookean is really a family of strain energy density functions. Our implementation should serve as a good starting point.

**Reply by rh201** [reply]
Thanks for pointing it out! @hugary1995 

I have a follow-up question: 

what's the difference between [ComputeFiniteStrain]+[StressDivergenceTensors] (with **use_displaced_mesh = true**) and [ComputeLagrangianStrain] + [TotalLagrangianStressDivergence] in MOOSE? I'm confused with the simple setting like
 **use_displaced_mesh = true** in [TensorMechanics/MasterAction].  Thank you.

**Reply by hugary1995** [reply]
You should not need to set this parameter if you are using the action. The strain calculators always run on the undisplaced mesh. The stress calculators run on the configuration where your properties are defined. The total Lagrangian kernels run on the undisplaced mesh. The incremental Lagrangian kernels run on the displaced mesh.
